soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx

```diff
@@ -1,14 +1,15 @@
 # ST Microelectronics STM32L010XX MCU
 #
 # Copyright (c) 2021 Wouter Cappelle <wouter.cappelle@crodeon.com>
+# Copyright (c) 2023 OS Systems
 # SPDX-License-Identifier: Apache-2.0
 
-if SOC_STM32L010XB
+if SOC_STM32L010XX
 
 config SOC
-	default "stm32l010xb"
+	default "stm32l010xx"
```

Comment:
This won't work, CONFIG_SOC has to match defines available hre: https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l0xx/soc/stm32l0xx.h#L61 (ie file names in https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l0xx/soc/)

Reply:
Is the NUM_IRQS realy 32? The RM show from 0-29 IRQ vectors.
Can you confirm?

Reply:
Looking to https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l0xx/soc/stm32l010x4.h#L96, it should be 30.

Reply:
I send a separated patch as last commit to fix it.